File: src/ipc/atlaspro.ts
import { invoke } from '@tauri-apps/api/core'

import type { AtlasProExecuteRequest, AtlasProReport, SpriteSource } from '../types'

export function scanAtlasProInputs(inputs: string[], recursive: boolean) {
  return invoke<SpriteSource[]>('scan_atlaspro_inputs', {
    request: { inputs, recursive },
  })
}

export function executeAtlasPro(request: AtlasProExecuteRequest) {
  return invoke<AtlasProReport>('execute_atlaspro', { request })
}

File: src-tauri/src/ipc/atlaspro.rs
use crate::error::AppError;
use crate::tools::atlaspro::model::{
    AtlasProExecuteRequest, AtlasProReport, AtlasProScanRequest, SpriteSource,
};

#[tauri::command]
pub fn scan_atlaspro_inputs(request: AtlasProScanRequest) -> Result<Vec<SpriteSource>, AppError> {
    crate::tools::atlaspro::scanner::scan(request)
}

#[tauri::command]
pub fn execute_atlaspro(request: AtlasProExecuteRequest) -> Result<AtlasProReport, AppError> {
    crate::tools::atlaspro::pipeline::execute(request)
}
